tests/test_misc.py

```diff
@@ -3093,3 +3093,54 @@ def test_checkmessage_pubkey_not_found(node_factory):
     check_result = l1.rpc.checkmessage(msg, zbase, pubkey=pubkey)
     assert check_result["pubkey"] == pubkey
     assert check_result["verified"] is True
+
+
+@pytest.mark.skip(reason="Fails by intention for creating test gossip stores")
+def test_create_gossip_mesh(node_factory, bitcoind):
+    """
+    Feel free to modify this test and remove the '@pytest.mark.skip' above.
+    Run it to get a customized gossip store. It fails on purpose, see below.
+
+    This builds a small mesh
+
+      l1--l2--l3
+      |   |   |
+      l4--l5--l6
+      |   |   |
+      l7--l8--l9
+    """
+    def connect_and_fund(node1, node2):
+        node1.connect(node2)
+        scid, _ = node1.fundchannel(node2, wait_for_active=False)
+        return scid
+
+    l1, l2, l3, l4, l5, l6, l7, l8, l9 = node_factory.get_nodes(9)
+    scid12 = connect_and_fund(l1, l2)
+    scid14 = connect_and_fund(l1, l4)
+    scid23 = connect_and_fund(l2, l3)
+    scid25 = connect_and_fund(l2, l5)
+    scid36 = connect_and_fund(l3, l6)
+    scid45 = connect_and_fund(l4, l5)
+    scid47 = connect_and_fund(l4, l7)
+    scid56 = connect_and_fund(l5, l6)
+    scid58 = connect_and_fund(l5, l8)
+    scid69 = connect_and_fund(l6, l9)
+    scid78 = connect_and_fund(l7, l8)
+    scid89 = connect_and_fund(l8, l9)
+    bitcoind.generate_block(6)
```

Comment:
Hm, tried that, I need to call `bitcoind.generate_bock()` after the calls to `fundchannel`. Even with `wait_for_active = False` some of the corresponding TX are already processed (by fundchannel calling bitcoind internally). Which then leads to an endless waiting, as not enough TX can be waited for (any more)

Reply:
No, I'm saying you can `fundchannel(... wait_for_active=False)` and then have `bitcoind` wait for all 12 funding txs to be in the mempool, and confirm them all at once. If you're still seeing interim blocks being generated, that may be due to `fundchannel` being magic, conditionally funding the node's wallet first, you'd have to pre-fund them before calling `fundchannel`, but the overall time to build the graph should reduce, since we're waiting once for 12 txs, rather than 12 times for 12 channels.

Reply:
Ah, so you want to speed up the 'wait for all nodes see the scids' cycle.
This is already quite quick, as it is parallel already.
The first one where it need to wait longer will make the next iterations faster as the other daemons are not sleeping in that time. 

When I used `wait_for_active=True` that is terribly slow as it does indeed one by one. But by using `wait_for_active=False` and cycling on all later

```py
    # waits for all nodes to have all scids gossip active
    for n in nodes:
        for scid in scids:
            n.wait_channel_active(scid)
```
Thats quite quick

Reply:
Also using  `wait_for_active=False` (as I do) makes it parallel on channel funding and the `wait_channel_active` loop below is somewhat parallel as I tried to explain.

Reply:
My main concern here is that we're wasting a lot of time in the `fundchannel` call by adding funds on demand to the nodes (causing `fundchannel` to advance the blockchain too), when we really could just use `lx.rpc.fundchannel` and speed things up. Notice that `lx.fundchannel` without the `rpc` is a wrapper that has some magic behavior which is what may be costing us time here:

https://github.com/ElementsProject/lightning/blob/e315f30728cbb1c259b27a943080f7c110366179/contrib/pyln-testing/pyln/testing/utils.py#L995-L996

Anyway, this is probably a premature optimization, so feel free to just mark this as resolved :-)

Reply:
I see what you mean, test would run faster if I pre-fund all nodes and wait_for_mempool, but also more complex in the test file.

Since this supposed to be only called for creating test data (and its not that slow) I tend to keep it as it is. Because otherwise I would be tempted to further optimize the code `contrib/pyln-testing/pyln/testing/utils.py` so that the resulting testing code is simpler.